mypy/stubgenc.py

```diff
@@ -508,12 +508,14 @@ def is_classmethod(self, class_info: ClassInfo, name: str, obj: object) -> bool:
             return inspect.ismethod(obj)
 
     def is_staticmethod(self, class_info: ClassInfo | None, name: str, obj: object) -> bool:
-        if self.is_c_module:
+        if class_info is None:
             return False
+        elif self.is_c_module:
+            raw_lookup = getattr(class_info.cls, "__dict__")  # noqa: B009
```

Comment:
To be honest, I wrote it that way because there are two other places it was used in two other places in stubgenc.py
- https://github.com/python/mypy/blob/8121e3c393ecd89e1d0e1ae98170bc2c46f7f1c3/mypy/stubgenc.py#L453-L455
- https://github.com/python/mypy/blob/8121e3c393ecd89e1d0e1ae98170bc2c46f7f1c3/mypy/stubgenc.py#L718-L725
I will check the file history to see if I can find a specific reason why `getattr` is used.

Reply:
The line originally comes from commit 6bcaf409 and was added to make typeshed happy:

https://github.com/python/mypy/blob/6bcaf40932e102df2a12c576e03c83089cce3d03/mypy/stubgenc.py#L140-L142